flybywiresim/aircraft
# fix(ecam):  fix ALL button behavior being inconsitent + cycling through STS page

Fixes #8888

## Summary of Changes


This PR stops the ALL ECAM button cycling through the STS page. It also fixes inconsistencies when pressing the button in quick succession and also makes sure it always starts from the ENG page if no page was selected.
Additionally the CRUISE Page is now skipped as well.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): \_chaoz_

## Testing instructions


1. Start however you want
2. Once SD is powered up, press the ALL button with singular clicks
3. SD should cycle through pages starting from ENG page till FCTL page and then repeat.
4. Deselecting a page should return to the flight phase appropriate page
5. Now hold the ALL button down
6. SD should cycle through pages starting from ENG page till FCTL page and then ***stop*** with a 3 second delay.
7. Deselecting a page should return to the flight phase appropriate page
8. Select any ECAM page
9. Pressing or holding the ALL button should cycle through pages starting from the selected page till it hits the FCTL page and then stop


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
